### DISCUSSION TITLE
What is the equivalent feature of useMultiStyleConfig in V3 and how can I pass props to the theme file?

### DISCUSSION THREAD

**Discussion by Pavi-thra-27** [chakra-ui/chakra-ui#9258] [Q&A] [answered]
For instance, 
`  const styles = useMultiStyleConfig('CustomButton', { isDisabled });`

What is the proposed way to implement the above one in V3?

**Comment by Kysluss** [comment]
I would also be curious in what the proposed solution is for this?  I have a component that used to use the `Wrap` component that would mimic the stylings of an `Input`.  I can swap out `Wrap` for `Box` but I cannot get the stylings to match up.  I can't use the `as` prop because an `Input` cannot have children.  I also cannot use `asChild` because there are other children other than the input (it's an auto-complete component with `Tag` components rendered for what has been selected).

**Comment by therrou** [comment]
Related question here: in v2, to create dynamic styles based on props, we could do something like:

```
const variantOutline: SystemStyleFunction = (props) => {
    const { colorScheme: c } = props;
    const base = {
        bg: "transparent",
        _disabled: {
            color: "black",
        },
        _active: {
            color: "black",
        },
    };

    if (c === "black") {
      ...rest of styles for props.colorScheme === "black"
      }
```

What would be the equivalent solution for v3?

EDIT: Just checked the migration docs, and based on this: https://chakra-ui.com/blog/00-announcing-v3#improve-runtime-performance 
"Remove runtime functions in theme: Previously we allowed the use of functions in the theming system which could negatively impact runtime performance. We've now switched to a variant-based system called "recipes", largely inspired by Panda CSS."
I guess we should write all the variants?

**Comment by segunadebayo** [comment, marked as answer]
v3 uses recipes instead of `useMultiStyleConfig`. https://chakra-ui.com/docs/theming/recipes
